2-mandatory/3-financial-times.js

```diff
@@ -23,59 +33,81 @@ function titleWithFewestWords(allArticleTitles) {
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+  let popularHeadline = [];
+  for (let i in allArticleTitles) {
+    if (allArticleTitles[i].match(/[0-9]/g) !== null) {
+      popularHeadline.push(allArticleTitles[i]);
+    }
+  }
+  return popularHeadline;
+  // TODO
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
-}
-
+  let aveNumofCharacters = [];
+  for (let i in allArticleTitles) {
+    aveNumofCharacters.push(allArticleTitles[i].length);
+  }
+  let total = 0;
+  for (let j in aveNumofCharacters) {
```

Comment:
I'd be careful of using single character variable names just because if you have code which is 50-100 lines long, it becomes hard to read. What you have done here is fine it's just something to consider.